app/models/manageiq/providers/openstack/inventory/parser/infra_manager.rb

```diff
@@ -0,0 +1,258 @@
+    hardware.bios                 = extra_attributes.fetch_path('firmware', 'bios', 'version')
+    # Can't get these 2 from ironic, maybe from Glance metadata, when it will be there, or image fleecing?
+    hardware.guest_os_full_name   = nil
+    hardware.guest_os             = nil
+    hardware.disks                = process_host_hardware_disks(hardware, extra_attributes)
+    hardware.introspected         = introspection_details.present?
+    # fog-openstack baremetal service defaults to Ironic API v1.1.
+    # In version 1.1 "available" is shown as null in JSON. It is correctly
+    # shown as "available" starting with version 1.2.
+    # This may need to change once this issue is addressed:
+    # https://github.com/fog/fog-openstack/issues/197
+    hardware.provision_state      = host.provision_state.nil? ? "available" : host.provision_state
+    
+    hardware
+  end
+
+  def get_extra_attributes(introspection_details)
+    return {} if introspection_details.blank? || introspection_details["extra"].nil?
+    introspection_details["extra"]
+  end
+
+  def process_host_hardware_disks(hardware, extra_attributes)
+    return [] if extra_attributes.nil? || (disks = extra_attributes.fetch_path('disk')).blank?
+    inventory_disks = []
+
+    disks.keys.delete_if { |x| x.include?('{') || x == 'logical' }.map do |d|
+      # Logical index contains number of logical disks
+      # TODO(lsmola) For now ignoring smart data, that are in format e.g. sda{cciss,1}, we need to design
+      # how to represent RAID
+      # Convert the disk size from GB to B
+      disk                 = persister.disks.find_or_build_by(:hardware => hardware, :device_name => d)
+      disk_size            = disks.fetch_path(d, 'size').to_i * 1_024**3
+      disk.device_type     = 'disk'
+      disk.controller_type = 'scsi'
+      disk.present         = true
+      disk.filename        = disks.fetch_path(d, 'id') || disks.fetch_path(d, 'scsi-id')
+      disk.location        = d
+      disk.size            = disk_size
+      disk.disk_type       = nil
+      disk.mode            = 'persistent'
+      inventory_disks << disk
+    end
+
+    inventory_disks
+  end
+
+  def clusters_and_host_mapping
+    clusters = []
+    cluster_host_mapping = {}
+    orchestration_stacks = collector.orchestration_stacks
+    orchestration_stacks&.each do |stack|
+      uid = stack.parent
+      next unless uid
+
+      nova_server = stack.resources.detect do |r|
+        collector.stack_server_resource_types.include?(r.resource_type)
+      end
+      next unless nova_server
+
+      host = collector.hosts.find { |h| h.instance_uuid == nova_server.physical_resource_id }
+      cluster_host_mapping[host.uuid] = uid
+
+      name = collector.orchestration_stacks.find { |s| s.id == stack.parent }.stack_name
+      clusters << {:name => name, :uid => uid}
+    end
+    return clusters.uniq, cluster_host_mapping
+  end
+
+  def find_resource(_uid, _stack_id)
```

Comment:
This is empty for infra manager, I wanted to avoid duplicating `orchestration_stack_resources` method.